modules/spring-config/src/main/java/org/mule/runtime/config/internal/SpringMuleContextServiceConfigurator.java

```diff
@@ -241,6 +243,7 @@
       .put(OBJECT_SCHEDULER_BASE_CONFIG, getBeanDefinition(SchedulerBaseConfigFactory.class))
       .put(OBJECT_CLUSTER_SERVICE, getBeanDefinition(DefaultClusterService.class))
       .put(LAZY_COMPONENT_INITIALIZER_SERVICE_KEY, getBeanDefinition(NoOpLazyComponentInitializer.class))
+      .put(METADATA_CACHE_MANAGER_KEY, getBeanDefinition(DefaultPersistentMetadataCacheManager.class))
```

Comment:
is this only to be able to inject things into instances? cant you use the muleContext's injector instead?

Reply:
I had to add this to the registry instead of registry-bootstrap mechanism as I need to change the implementation of the service if a shared persistence folder for OS is set on lazy mule artifact contexts.